manifests/bootstrap_rpm.pp

```diff
@@ -0,0 +1,66 @@
+# This class builds an RPM containing the bootstrap for a subscription-manager consumer
+# This file is placed in $katello_www_pub_dir.
+# @api private
+class foreman_proxy_content::bootstrap_rpm (
+  String $hostname = $facts['networking']['fqdn'],
+  String $deployment_url = '/rhsm',
+  Stdlib::Port $rhsm_port = 443,
+  Stdlib::Absolutepath $rhsm_ca_dir = '/etc/rhsm/ca',
+  String $candlepin_cert_rpm_alias_filename = 'katello-ca-consumer-latest.noarch.rpm',
+  Stdlib::Absolutepath $katello_www_pub_dir = '/var/www/html/pub',
+) {
+  include certs
+
+  $katello_server_ca_cert = $certs::katello_server_ca_cert,
+  $server_ca_name = $certs::server_ca_name,
+  $default_ca_name = $certs::default_ca_name,
+  $ca_cert = $certs::ca_cert,
+  $server_ca = $certs::server_ca,
+
+  $katello_rhsm_setup_script = 'katello-rhsm-consumer'
+  $katello_rhsm_setup_script_location = "/usr/bin/${katello_rhsm_setup_script}"
+
+  $candlepin_consumer_name = "katello-ca-consumer-${hostname}"
+  $candlepin_consumer_summary = "Subscription-manager consumer certificate for Katello instance ${hostname}"
+  $candlepin_consumer_description = 'Consumer certificate and post installation script that configures rhsm.'
+
+  include trusted_ca
+  trusted_ca::ca { 'katello_server-host-cert':
+    source  => $katello_server_ca_cert,
+    require => File[$katello_server_ca_cert],
+  }
```

Comment:
This is also a part I always wondered about: why does the server need to globally trust the server certificate? Don't we set the CA explicitly everywhere?

Reply:
Looks like, based on https://projects.theforeman.org/issues/13634, we could drop it for the reasons stated it was added. However, there is currently known code that does rely on this we would need to fix. I was discussing that last week with @jlsherrill

Reply:
Perhaps we can start with adding a class parameter `Boolean $add_ca_as_trusted_ca = true` so we can toggle it to false in nightly to see what breaks? Since it's a private class, users shouldn't see anything.

Reply:
I know at least Pulp communication will likely break due to how that is currently coded. Let me compromise with creating a dedicated issue to explore this (https://projects.theforeman.org/issues/30338) so that this code is still as clean as can be a copy from puppet-certs. Then it can be a targeted follow up along with the bootstrap RPM being re-factored to drop katello-cert-tools as a stand alone fix.

I do agree our code should be robust against the CA certificate being in the system wide truststore but are there other reasons that we shouldn't do this?

Reply:
The less invasive we are to the system, the easier. We can also drop the `trusted_ca` module from the installer. Right now I'm maintaining that in the Voxpupuli organization because we need it. That overhead might not be directly visible, but it's there.

I also don't know if there may be security policies on customer systems that would disallow this. Perhaps now they don't know so don't care, but might if they did.

Let's keep this and make it part of the "kill pulp 2"-PR